### PR TITLE
[SPARK-16140][MLlib][SparkR][Docs] Group k-means method in generated R doc

### PR DESCRIPTION
**Pull request by keypointt**
https://issues.apache.org/jira/browse/SPARK-16140
## What changes were proposed in this pull request?

Group the R doc of spark.kmeans, predict(KM), summary(KM), read/write.ml(KM) under Rd spark.kmeans. The example code was updated.
## How was this patch tested?

Tested on my local machine

And on my laptop `jekyll build` is failing to build API docs, so here I can only show you the html I manually generated from Rd files, with no CSS applied, but the doc content should be there.

screenshotkmeans

### CODE REVIEW

**Comment by mengxr** [comment]
@junyangq Could you help review this PR? Thanks!

**Comment by junyangq** [comment]
It would be better to group write.ml (KM) into the doc as well. Also please include a full screenshot of the doc page.

File: R/pkg/R/mllib.R [hunk 1]
```diff
@@ -277,14 +277,32 @@ setMethod("summary", signature(object = "NaiveBayesModel"),
...
+#' model <- spark.kmeans(df, Sepal_Length ~ Sepal_Width, k = 4, initMode = "random")
+#' summary(model)
+#'
+#' # fitted values on training data
+#' fitted <- predict(model, df)
+#' head(select(fitted, "Sepal_Length", "prediction"))
+#'
+#' # save fitted model to input path
+#' path <- "path/to/model"
+#' write.ml(model, path)
```
Comment (keypointt) [inline R/pkg/R/mllib.R]: hi @junyangq , isn't it `write.ml (KM)` you are talking about?

**Comment by keypointt** [comment]
And on my laptop `jekyll build` is failing to build API docs, so here I can only show you the html I manually generated from Rd files, with no CSS applied, but the doc content should be there.

screen shot 2016-06-27 at 1 04 36 pm

File: R/pkg/R/mllib.R (hunk 1 above)
Comment (junyangq) [inline R/pkg/R/mllib.R]: Yes, can you modify the doc of write.ml method for KM to bring it to the same page as other KM methods?

Comment (keypointt) [inline R/pkg/R/mllib.R]: oh sorry I see what you mean now, you mean bring it to the `Usage` section of this page, right? I'll try to do it

**Comment by keypointt** [comment]
I just updated the screenshot in PR description

**Comment by junyangq** [comment]
That looks better now. Two more comments
- the description section can be made more compact as in #13877 
- duplicated arguments should be deleted as well

**Comment by keypointt** [comment]
could you please be more specific on `compact `? like what should be changed then? I'm not getting it completely

I'm removing the duplicate arguments now

File: R/pkg/R/mllib.R [hunk 2]
```diff
@@ -266,9 +266,9 @@ setMethod("summary", signature(object = "NaiveBayesModel"),
             return(list(apriori = apriori, tables = tables))
           })
 
-#' Fit a k-means model
+#' K-Means Model
```
Comment (mengxr) [inline R/pkg/R/mllib.R]: `K-Means Clustering Model`

**Comment by mengxr** [comment]
@keypointt For quicker tests, you can run `R/create-docs.sh` and then check the html doc under `R/pkg/html`. It would be much faster than `jekyll build`.

File: R/pkg/R/mllib.R (hunk 2 above)
Comment (junyangq) [inline R/pkg/R/mllib.R]: sorry I actually meant the description could be more like plain sentences, rather than bullet points.

Comment (mengxr) [inline R/pkg/R/mllib.R]: This appears as the title, like in `Generalized Linear Models`.

**Comment by keypointt** [comment]
@junyangq oh I got it thanks
@mengxr I just realized the `jekyll build ` is just copying over whatever under `R/pkg/html/`, thanks for the tip

**Comment by mengxr** [comment]
I just merged #13927. So please fetch and merge the latest master if you want to push an update to avoid merge conflict (hopefully none).

**Comment by keypointt** [comment]
OK doing it now

**Comment by mengxr** [comment]
I think the error was because this PR left `predict`, `write.ml`, etc documented without title. So this PR has to be combined with SPARK-16144. Basically, let us add some doc to the function declarations under `generics.R`.

cc: @yinxusen

**Comment by keypointt** [comment]
hi @mengxr I tried to modifiy in generics.R

```
#' @rdname spark.kmeans
#' @export
setGeneric("spark.kmeans", function(data, formula, ...) { standardGeneric("spark.kmeans") })
```
tried many different ways but still not working, since I'm not familiar with R docs, maybe here I need your help so that it's not blocking the other tickets.

How to add `Sections \title, and \name `? I guess it's because both algorithms are trying to define a `predict` or `write.ml`

File: R/pkg/R/mllib.R [hunk 3]
```diff
@@ -358,19 +369,11 @@ setMethod("summary", signature(object = "KMeansModel"),
                    cluster = cluster, is.loaded = is.loaded))
           })
 
-#' Predicted values based on model
-#'
-#' Makes predictions from a k-means model or a model produced by spark.kmeans().
+#  Predicted values based on a k-means model
 #'
-#' @param object A fitted k-means model
-#' @rdname predict
```
Comment (keypointt) [inline R/pkg/R/mllib.R]: or maybe we can just remove it from R docs? anyway predict should be a separate generic method across all ml algorithmns

File: R/pkg/R/mllib.R [hunk 4]
```diff
@@ -477,24 +480,15 @@ setMethod("write.ml", signature(object = "GeneralizedLinearRegressionModel", pat
...
-#'
-#' Save the k-means model to the input path.
+#  Save fitted MLlib model to the input path
 #'
-#' @param object A fitted k-means model
 #' @param path The directory where the model is saved
 #' @param overwrite Overwrites or not if the output path already exists. Default is FALSE
 #'                  which means throw exception if the output path exists.
 #'
-#' @rdname write.ml
```
Comment (keypointt) [inline R/pkg/R/mllib.R]: write.ml same as predict

**Comment by yinxusen** [comment]
@keypointt You need to add title for `predict` and `write.ml`. Like the first line below.

```Rd
#' This is title for write.ml
#' @rdname write.ml
#' @export
setGeneric("write.ml", function(object, path, ...) { standardGeneric("write.ml") })
```

**Comment by keypointt** [comment]
oh I see, thanks a lot

**Comment by felixcheung** [comment]
Actually don't do this? 

```
#' This is a title for write.ml
#' @rdname write.ml
#' @export
setGeneric("write.ml", function(object, path, ...) { standardGeneric("write.ml") })
```

it will create a dummy page with no content for write.ml. And generally we don't put more than @rdname and @export in generics.R - it is harder to manage from there.

I also was thinking about the discoverability of "write.ml", "predict" and so on in the doc. I think it is better to have full page for "write.ml" and then link to write.ml(KM), write.ml(GLM) etc from there.

**Comment by keypointt** [comment]
I vote for dis-coverability of "write.ml", "predict"

**Comment by felixcheung** [comment]
And to elaborate, the reason why it is complaining is because the line in generics.R is the only reference to @rdname write.ml. To fix, you could either change that to @rdname spark.kmeans (which doesn't make sense because there are multiple implementation of write.ml for different model), or have an actual @rdname write.ml block in mllib.R - this also helps to redirect to pages on other mllib models in the event someone is looking for write.ml and doesn't know to look in spark.kmeans for write.ml.

**Comment by keypointt** [comment]
if we are to have an actual @rdname write.ml block in mllib.R, then it has to be enforced to all documentation stories under https://issues.apache.org/jira/browse/SPARK-16090, to remove individual `write.ml` and `predict` of each model

**Comment by mengxr** [comment]
@keypointt Could you add `seealso` links to the generic `predict` and `write.ml` doc?

@felixcheung I think it is useful to have a page for the generic `predict` and `write.ml`. We just need to add some links to them so they do provide some information. Btw, is it necessary to define `predict` under `generics.R`? Is it a common practice in R packages?

**Comment by keypointt** [comment]
sure, I tried 
```
#' @seealso \link{predict}, \link{read.ml}, \link{write.ml}
```

and it's like below for `spark.kmeans` page, is this what you mean?
screen shot 2016-06-28 at 1 21 25 pm


But if I remove the previously added `title` in generics.R, then errors still pop up

**Comment by mengxr** [comment]
The grouping of k-means methods LGTM. So I'm merging this into master and branch-2.0 and leave the remaining issues to SPARK-16144. Thanks!

**Pull request closed without merging**